Makefile

```diff
@@ -72,6 +75,20 @@ releases:
 .PHONY: package
 package: clean releases
 
+.PHONY: publish-development
+publish-development: push-to-charmstore
+	$(CHARMCMD) publish --channel development $(fullurl)
+
+.PHONY: publish-stable
+publish-stable: push-to-charmstore
+	$(CHARMCMD) publish $(fullurl)
+
+.PHONY: push-to-charmstore
+push-to-charmstore: clean
+	$(CHARMCMD) login
+	@echo $(CHARMCMD) push . $(CHARMURL)
+	$(eval fullurl = $(shell $(CHARMCMD) push . $(CHARMURL) | grep 'url: ' | awk -F 'url: ' '{print $$2}'))
```

Comment:
This step is idempotent, correct? An unchanged charm will return the same url as the previous time it was pushed?